### PR TITLE
Deploy Provider: List only Active Task Definitions

### PR DESCRIPTION
**Pull request by jparsons04**
**What does this pull request do?**
Deploy List() will now explicitly only return active Task Definitions and Task Definition Families.

**How should this be tested?**
Manually test Deploy methods with Swagger
Unit tests

**Checklist**
- [x] Unit tests
~- [ ] Smoke tests (if applicable)~
~- [ ] System tests (if applicable)~
~- [ ] Documentation (if applicable)~

### CODE REVIEW

File: api/controllers/deploy.go [hunk 1]
```diff
@@ -70,7 +70,13 @@ func (d *DeployController) GetDeploy(c *fireball.Context) (fireball.Response, er
 }
 
 func (d *DeployController) ListDeploys(c *fireball.Context) (fireball.Response, error) {
-	summaries, err := d.DeployProvider.List()
+	var all bool
+
+	if c.PathVariables["all"] != "" {
+		all = true
```
Comment (zpatrick) [inline api/controllers/deploy.go]: Lets use https://golang.org/pkg/strconv/#ParseBool

File: api/controllers/deploy_test.go [hunk 2]
```diff
@@ -120,7 +121,7 @@ func TestListDeploys(t *testing.T) {
 	}
 
 	mockDeployProvider.EXPECT().
-		List().
+		List(all).
```
Comment (zpatrick) [inline api/controllers/deploy_test.go]: You need to add this query param to the fireball.Context (see task logs test) and test when all=true

File: api/provider/aws/deploy_list.go [hunk 3]
```diff
@@ -34,8 +36,12 @@ func (d *DeployProvider) listTaskDefinitionARNs() ([]string, error) {
 	familyPrefix := addLayer0Prefix(d.Config.Instance(), "")
 	input := &ecs.ListTaskDefinitionFamiliesInput{}
 	input.SetFamilyPrefix(familyPrefix)
-	// TODO: Revisit how Inactive and Active Task Definitions might want to be returned to the client
-	input.SetStatus(ecs.TaskDefinitionFamilyStatusActive)
+	if all {
```
Comment (zpatrick) [inline api/provider/aws/deploy_list.go]: [nit] try to void if/else blocks. Set status active and then only have the “if” block for when all=true

File: api/controllers/deploy_test.go (hunk 2 above)
Comment (jparsons04) [reply api/controllers/deploy_test.go]: :white_check_mark: 

I made this adjustment and added a new test, but I've added a note in the test that the unit test can't actually test properly filtering out Active/Inactive Task Definitions and would have to be done through a system test. As it stands, we're just testing the controller's ability to handle all=false and all=true.

**Comment by jparsons04** [comment]
Something I'm concerned about with this implementation is the fact that we are allowing a client to list all inactive Task Definitions and Task Definition Families, but List() currently will only return those Deploy Summaries that [have a match in the tags db](https://github.com/quintilesims/layer0/blob/api-refactor-deploy-provider-list-all-jparsons/api/provider/aws/deploy_list.go#L90).

In my specific case, I have about 11k inactive Task Definitions (this is normal per [AWS's note on this doc page](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DeregisterTaskDefinition.html)), so when I do an `all=true` query for List, the process for me can take upwards of 30 seconds, but then I get back a single Deploy that I created through Swagger.

**Comment by jparsons04** [comment]
In this case, no, because we're not including the new query parameter, so the swagger controller doesn't change.

```             "/deploy": map[string]swagger.Method{
                 "get": {
                     Summary: "List all Deploys",
                     Tags:    []string{"Deploy"},
                     Responses: map[string]swagger.Response{
                         "200": {
                             Description: "An array of deploys",
                             Schema:      swagger.NewObjectSliceSchema("Deploy"), 
                         },
                     },
                 },```

**Comment by jparsons04** [comment]
Likewise, the Deploy controller doesn't change because it's making the same call to `DeployProvider.List()` as it always has.

**Pull request merged**